src/lib/OpenEXRUtil/ImfCheckFile.cpp

```diff
@@ -636,6 +637,39 @@ readDeepTile(T& in,bool reduceMemory , bool reduceTime)
     return threw;
 }
 
+//
+// EXR will read files that have out-of-range values in certain enum attributes, to allow
+// values to be added in the future. This function returns 'false' if any such enum attributes
+// have unknown values
+//
+// (implementation node: it is undefined behavior to set an enum variable to an invalid value
+//  this code circumvents that by casting the enums to integers and checking them that way)
+//
+bool enumsValid( const Header& hdr)
```

Comment:
I understand this part of the change and it makes sense to me, for the record :) I do wonder if it would be resolved by

```
enum Envmap : int 
```
though